### PR TITLE
overlaP using portion package

### LINKED ISSUE #3: Overlap function
Update to use portion package

### PR DESCRIPTION
**Pull request by tseamuscorlett**
1) added overlaP function that uses portion package. overlaP returns True if there's an overlap. I'm keeping the old overlap function which returns an integer (length of overlap) because I don't think portion has that method (please check portion documentation).

2) Also added a test.py for unit testing!

### CODE REVIEW

**Comment by liamlongo5** [comment]
Per portion documentation:

>>> P.closed(0, 2) & P.closed(1, 3)
[1,2]
>>> P.closed(0, 4) & P.open(2, 3)
(2,3)
>>> P.closed(0, 2) & P.closed(2, 3)
[2]
>>> P.closed(0, 2) & P.closed(3, 4)
()

File: main.py [hunk 1]
```diff
@@ -123,6 +125,14 @@ def overlap(range1, range2):
     else:
         return 0
 
+def overlaP(range1, range2):
```
Comment (liamlongo5) [inline main.py:128]: overlap and overlaP is a recipe for confusion

File: main.py [hunk 2]
```diff
@@ -123,6 +125,14 @@ def overlap(range1, range2):
...
+    '''
+    Return True if two ranges overlap, False otherwise.
+    Uses portion package.
+    '''
+    return P.closed(range1[0], range1[1]).overlaps(P.closed(range2[0], range2[1]))
```
Comment (liamlongo5) [inline main.py:133]: I left you a comment on this commit that shows how you can use portion package for calculating the length of the interval

File: test.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import unittest
```
Comment (liamlongo5) [inline test.py:1]: Unit tests!
Beautiful!

File: main.py (hunk 2 above)
Comment (tseamuscorlett) [reply main.py:133]: Just started working on overlapBool, overlapLength and overlapRange

**Review by liamlongo5 — APPROVED** [review]
couple minor things.
I suggest you get portion up and running rather than having two functions :)

File: main.py (hunk 2 above)
Comment (tseamuscorlett) [reply main.py:133]: ALL using portion package

**Pull request merged**